handlers/user.go

```diff
@@ -1298,50 +1322,86 @@ func UserUpdateEmailPost(w http.ResponseWriter, r *http.Request) {
 		http.Redirect(w, r, "/user/settings", http.StatusSeeOther)
 		return
 	}
-	email := r.FormValue("email")
 
-	if !utils.IsValidEmail(email) {
-		session.AddFlash("Error: Invalid email format!")
+	// check if password is correct
+	formPassword := r.FormValue("password")
+
+	err = bcrypt.CompareHashAndPassword([]byte(userData.Password), []byte(formPassword))
+	if err != nil {
+		session.AddFlash("Error: Invalid email or password!")
 		session.Save(r, w)
 		http.Redirect(w, r, "/user/settings", http.StatusSeeOther)
 		return
 	}
 
-	var existingEmails struct {
-		Count int
-		Email string
+	// validate new email
+	newEmail := r.FormValue("email")
+
+	if userData.Email == newEmail {
+		session.Save(r, w)
+		http.Redirect(w, r, "/user/settings", http.StatusSeeOther)
+		return
 	}
-	db.FrontendWriterDB.Get(&existingEmails, "SELECT email FROM users WHERE email = $1", email)
 
-	if existingEmails.Email == email {
+	if !utils.IsValidEmail(newEmail) {
+		session.AddFlash("Error: Invalid email format!")
+		session.Save(r, w)
 		http.Redirect(w, r, "/user/settings", http.StatusSeeOther)
 		return
-	} else if existingEmails.Email != "" {
-		session.AddFlash("Error: Email already exists please choose a unique email")
+	}
+
+	emailExists := false
+	db.FrontendWriterDB.Get(&emailExists, "EXISTS (SELECT email FROM users WHERE email = $1)", newEmail)
```

Comment:
**issue:** This does not appear to be valid syntax. I can only find uses of `EXISTS` in the context of `WHERE EXISTS` online.

Furthermore there is no error handling for this line.

Reply:
fixed